src/app/project-sets/[projectSetId]/(components)/sidebarProjectsDisplay.tsx

```diff
@@ -0,0 +1,60 @@
+'use client'
+
+import * as React from "react"
+
+import {SearchBar} from "@/components/SearchBar"
+import {Button} from "@/components/ui/button"
+import {type Project} from "@/_temp_types/projects"
+import {useRouter} from "next/navigation"
+
+export type SidebarProjectsDisplayProps = {
+    projects: Project[]
+    currentSearchTerm: string
+    currentProjectId: number
+    currentEditMode: boolean
+}
+
+export function SidebarProjectsDisplay({projects, currentSearchTerm, currentProjectId, currentEditMode}: SidebarProjectsDisplayProps) {
+    const router = useRouter()
+
+    const updateSearchTerm = (newSearchTerm: string) => {
+        router.push(`?isEdit=${currentEditMode}&projectId=${currentProjectId}&search=${newSearchTerm}`)
+    }
+
+    const updateProjectIdx = (newProjectId: number) => {
+        router.push(`?isEdit=${currentEditMode}&projectId=${newProjectId}&search=${currentSearchTerm}`)
+    }
+
+    const handleProjectChanged = (project: Project) => {
+        updateProjectIdx(project.id)
+        // TODO: shoot update api
+    }
+
+    return (
+        <>
+            <SearchBar
```

Comment:
UX-wise, it'd be nice if the search would dynamically filter as you type, but I know the whole search param thing makes that weird - even more reason to rethink how we're doing this